### ISSUE TITLE
Install xtermjs from npm

### ISSUE THREAD

**Issue by cancan101** [jupyter/notebook#3180]
Repost from https://github.com/jupyter/notebook/issues/2850#issuecomment-355616270:

New version of xterm (v3.0.1) [has been released](https://github.com/xtermjs/xterm.js/releases/tag/3.0.1). That being said, they have [dropped support for bower](https://xtermjs.org/docs/guides/download/#no-more-bower-support). Any chance to pull in dependencies from npm / yarn?
  
Looks like until this [PR](https://github.com/jupyter/notebook/pull/2011/files#diff-0a08a7565aba4405282251491979bb6bR21), xterm was loaded from npm.

**Comment by takluyver** [comment]
Do we have any Javascript dependencies installed with npm already? If so, feel free to move it over. If not, it's probably going to be a pain.

**Comment by cancan101** [comment]
I'm not an expert in front end development, but I do see: https://github.com/jupyter/notebook/blob/27c00bfb20f51356613a32dd01d00b0abf30544f/package.json#L18-L19

**Comment by takluyver** [comment]
I think those are dependencies for developers, not ones that get bundled to use in the frontend (requirejs appears both there and in bower.json).

**Comment by cancan101** [comment]
You may right. So what to do about xtermjs going forward given no longer publishing to bower? Any reason to prefer using bower vs npm (given the relative popularity of the two systems)?

**Comment by blink1073** [comment]
The reason for the reversal in #2011 is that the use of npm and Webpack is at odds with the ability to use `requirejs()` with specific modules - Webpack wants to bundle the modules and keep them internal.  This is a capability used by many existing notebook extensions.  The unfortunate result is that the classic notebook is tied to bower, which is slowly being phased out by the community in favor of WebPack and similar bundling tools.

**Comment by taion** [comment]
Those are somewhat orthogonal, right? You're pulling a lot of your Bower deps from unpkg anyway, which indicates that the UMD builds are actually published through npm.

Separately, and I'm sure you already know this, you should probably move off AMD at some point, since AMD is just as dead as Bower... might be a roadmap item given the legitimate (but rare) use thereof here.

**Comment by blink1073** [comment]
True, the real issue is that more packages (like xterm), are publishing commonjs or ES6+ instead of UMD/AMD.  If xterm were to publish a UMD/AMD package (as it previously did in its dist folder), it could be installed using npm and copied to the same folder as the bower packages.

**Comment by taion** [comment]
`xterm` still does. @cancan101 communicated to me verbally that he will submit a PR to use https://unpkg.com/xterm@3.0.1/dist/

**Comment by blink1073** [comment]
Ah, yep, I confirmed that locally.  We would still need to get the CSS file though...
